ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh

```diff
@@ -338,19 +338,30 @@ function check_latest_machineconfig_applied() {
 }
 
 function wait_machineconfig_applied() {
-    local role="${1}" try=0 interval=60
-    num=$(oc get node --no-headers -l node-role.kubernetes.io/"$role"= | wc -l)
-    local max_retries; max_retries=$(expr $num \* 20)
-    while (( try < max_retries )); do
-        echo "Checking #${try}"
-        if ! check_latest_machineconfig_applied "${role}"; then
-            sleep ${interval}
-        else
-            break
+    local role="${1}" try=0 interval=30
+    num=$(oc get node --no-headers -l node-role.kubernetes.io/"$role"= | wc -l) 
+    local max_retries; max_retries=$(expr $num \* 20 \*60 \/ $interval) # Wait 30 minutes for each node, try 60/interval times per minutes
+
+    local mcp_try=0 mcp_status=''
+    local max_try_between_updated_and_updating; max_try_between_updated_and_updating=$(expr 5 \*60 \/ $interval) # We consider mcp to be updated if its status is updated for 5 minutes
+    while [ $mcp_try -lt $max_try_between_updated_and_updating ] && [ $try -lt $max_retries ]
+    do
+        sleep ${interval}
+        echo "Checking MCP #${try}"
+        mcp_status=$(oc get mcp/$role -o json | jq -r '.status.conditions[] | select(.type == "Updated") | .status')
+        echo "MCP updated is ${mcp_status}"
```

Comment:
how about move it out of the loop to decrease the log info? it should be ok to output the $role mcp's status once after the loop for debug.

Reply:
sure